dev/io.openliberty.microprofile.telemetry.2.0.internal/src/io/openliberty/microprofile/telemetry20/internal/http/MPTelemetryHTTPMetricsAdapterImpl.java

```diff
@@ -79,22 +89,22 @@ public void updateHttpMetrics(HttpStatAttributes httpStatAttributes, Duration du
     private Attributes retrieveAttributes(HttpStatAttributes httpStatAttributes) {
 
         AttributesBuilder attributesBuilder = Attributes.builder();
-        attributesBuilder.put(OpenTelemetryConstants.HTTP_REQUEST_METHOD, httpStatAttributes.getRequestMethod());
-        attributesBuilder.put(OpenTelemetryConstants.URL_SCHEME, httpStatAttributes.getScheme());
+        attributesBuilder.put(HTTP_REQUEST_METHOD, httpStatAttributes.getRequestMethod());
+        attributesBuilder.put(URL_SCHEME, httpStatAttributes.getScheme());
 
-        Integer status = httpStatAttributes.getResponseStatus().orElse(-1);
-        attributesBuilder.put(OpenTelemetryConstants.HTTP_RESPONSE_STATUS_CODE, status == -1 ? "" : status.toString().trim());
+        Long status = Long.valueOf(httpStatAttributes.getResponseStatus().orElse(-1));
+        attributesBuilder.put(HTTP_RESPONSE_STATUS_CODE, status);
 
-        attributesBuilder.put(OpenTelemetryConstants.HTTP_ROUTE, httpStatAttributes.getHttpRoute().orElse(""));
+        attributesBuilder.put(HTTP_ROUTE, httpStatAttributes.getHttpRoute().orElse(""));
```

Comment:
Not part of the change here, but should we be setting the route attribute if we don't have the data for it?

Reply:
Good question, I'll chase it up.